### PR TITLE
[Infra] Remove `HeartbeatLoggingTestUtils.podspec`

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
Alternative approach to publishing `HeartbeatLoggingTestUtils.podspec`. The pod's sources make minimal use of XCTest APIs and can instead be replaced with non-XCTest code. The approach is to move the test util code into FirebaseCoreInternal and `#if DEBUG` it out (under the assumption that tests won't be run in `Release` mode.).

Tested some targets locally but letting CI handle the rest.

#no-changelog

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1355,5 +1346,5 @@ func googleAppMeasurementDependency() -> Package.Dependency {
     return .package(url: appMeasurementURL, branch: "main")
   }
 
-  return .package(url: appMeasurementURL, exact: "10.10.0")
+  return .package(url: appMeasurementURL, branch: "main")
```
Comment (paulb777) [inline Package.swift:1349]: Not this?

Comment (ncooke3) [reply Package.swift:1349]: correct. not sure how that I did that.

Comment (ncooke3) [reply Package.swift]: oh, i ran `scripts/setup_spm_test.sh` at one point. good catch!

File: FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatLoggingTestUtils.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 Google LLC
```
Comment (ncooke3) [inline FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatLoggingTestUtils.swift:1]:
```suggestion
// Copyright 2023 Google LLC
```

Comment (ncooke3) [reply FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatLoggingTestUtils.swift:1]: Hmm, does a renamed file need an updated copyright year? @paulb777

**Review by paulb777 — APPROVED** [review]

File: FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatLoggingTestUtils.swift (hunk 2 above)
Comment (paulb777) [reply FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatLoggingTestUtils.swift:1]: Nope, it can keep the 2021

**Review by andrewheard — APPROVED** [review]

**Pull request merged**